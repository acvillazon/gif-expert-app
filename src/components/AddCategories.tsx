import { useState } from "react";
import PropTypes from "prop-types";

interface Target {
  value: string;
}

export const AddCategories = ({ onNewCategories }: { onNewCategories: Function }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }: { target: Target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event: any) => {
    event.preventDefault();
    if (!inputValue.trim()) return;
    onNewCategories(inputValue.trim());
    setInputValue("");
  };

  return (
    <form aria-label="form" onSubmit={onSubmit}>
      <div className="mb-4 mt-4">
        <label className="block text-gray-700 text-sm font-bold mb-2">Category</label>
        <input
          type="text"
          className="block w-50 rounded-md border-0 py-1.5 pl-4 pr-20 text-gray-900 ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:outline-lime-300 focus:ring-green-100 sm:text-sm sm:leading-6"
          placeholder="Search Gift"
          value={inputValue}
          onChange={onInputChange}
        />
      </div>
    </form>
  );
};

AddCategories.propTypes = {
  onNewCategories: PropTypes.func.isRequired
}