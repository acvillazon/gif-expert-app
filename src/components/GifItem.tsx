import PropTypes from "prop-types";
import { Images } from "../interfaces/images";

export const GifItem = ({id, title, url}:Images) => {
  return (
    <div className="max-w-sm rounded overflow-hidden shadow-lg">
        <img className="object-cover h-48 w-full" alt={title} src={url}></img>
        <p className="italic text-center font-semibold" key={id}>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
