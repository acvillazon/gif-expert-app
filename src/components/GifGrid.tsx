import { Images } from '../interfaces/images'
import { GifItem } from './GifItem'
import { useFetchGifs } from '../hooks/useFetchGifs'
import PropTypes from 'prop-types';

export const GifGrid = ({ category }: any) => {
    const { images, isLoading } = useFetchGifs(category)

    return (
        <>
            <h2 className="font-bold text-xl mt-2 mb-2">{category}</h2>
            {
              isLoading && (<h1>Cargando!!</h1>)
            }
            <div className="gap-4 grid sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4">
                {images.map((img: Images) => (
                    <GifItem key={img.id} {...img} />
                ))}
            </div>
        </>
    )
}

GifGrid.propTypes ={
    category: PropTypes.string.isRequired
}
