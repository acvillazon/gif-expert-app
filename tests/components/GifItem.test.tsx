import React from "react";
import { render, screen } from "@testing-library/react"
import { GifItem} from '../../src/components/GifItem'

describe('GifItem tests', () => { 
    test('should take an snapshot of the component', () => { 
        const title = 'Andres';
        const url = 'URL';
        const id = '19827';
        const {container} = render(<GifItem title={title} url={url} id={id}/>)
        expect(container).toMatchSnapshot();
    })

    test('should display the URL of the image with its respective ALT', () =>{
        const title = 'Andres';
        const url = 'http://andres.villazon/test';
        const id = '19827';
        render(<GifItem title={title} url={url} id={id}/>)
        const {src, alt}: HTMLImageElement = screen.getByRole('img');
        expect( src ).toBe(url);
        expect( alt ).toBe(title);
    });

    test('should display the title passed', () =>{
        const title = 'Andres';
        const url = 'http://andres.villazon/test';
        const id = '19827';
        render(<GifItem title={title} url={url} id={id}/>)
        const titleSearch  = screen.getByText(title);
        expect( titleSearch ).toBeTruthy();
    });
});