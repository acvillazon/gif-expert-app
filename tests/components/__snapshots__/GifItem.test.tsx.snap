// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem tests should take an snapshot of the component 1`] = `
<div>
  <div
    class="max-w-sm rounded overflow-hidden shadow-lg"
  >
    <img
      alt="Andres"
      class="object-cover h-48 w-full"
      src="URL"
    />
    <p
      class="italic text-center font-semibold"
    >
      Andres
    </p>
  </div>
</div>
`;
